app/importers/reading/somerville_mega_reading_importer.rb

```diff
@@ -1,67 +1,90 @@
+# For importing reading data from Google Sheets or a local folder of csvs.
 class SomervilleMegaReadingImporter
-  def initialize(options:)
-    @log = options.fetch(:log)
-    @school_local_ids = options.fetch(:school_scope, [])
-    @file_text = options.fetch(:file_text, nil)
-
-    # TODO add dependencies as instance variables
-    # @student_ids_map = ::StudentIdsMap.new
+  def initialize(school_year, options = {})
+    @school_year = school_year
+    @log = options.fetch(:log, Rails.env.test? ? LogHelper::Redirect.instance.file : STDOUT)
     # @matcher = options.fetch(:matcher, ImportMatcher.new)
-    @syncer = ::RecordSyncer.new(log: @log)
-    reset_counters!
+    # @syncer = ::RecordSyncer.new(log: @log)
+    @fetcher = options.fetch(:fetcher, GoogleSheetsFetcher.new)
   end
 
   def import
-    streaming_csv = read_or_fetch_csv
-    if streaming_csv.nil?
+    streaming_csvs = read_or_fetch_sheet
+    if streaming_csvs.nil?
       log('Aborting since no CSV found...')
       return
     end
 
-    # TODO do any initialization before the core loop
-    # log('Building student_ids_map...')
-    # @student_ids_map.reset!
-    # log("@student_ids_map built with #{@student_ids_map.size} local_id keys")
-
     log('Starting loop...')
-    reset_counters!
-    streaming_csv.each_with_index do |row, index|
-      import_row(row)
-      log("processed #{index} rows.") if index % 1000 == 0
+    @valid_student_names_count = 0
+    @invalid_student_name_count = 0
+    @valid_data_points_count = 0
+
+    processor = MegaReadingProcessor.new(read_uploaded_by_educator_from_env, @school_year)
+    streaming_csvs.each_with_index do |tab, index|
+      rows, meta = processor.process(tab.tab_csv)
+      log("processed #{index} sheets.")
+      rows.each_with_index {|row, index| import_row(row, index)}
+      valid_student_names_count, valid_data_points_count, invalid_student_name_count = meta.values_at(
```

Comment:
I don't think you need to pull these metrics out here, the processor class already logs these stats: https://github.com/studentinsights/studentinsights/blob/master/app/importers/reading/mega_reading_processor.rb#L43 - but we'll need to pass the `log` instance along when creating the processor instance.